app/src/main/java/org/p2p/wallet/history/interactor/HistoryTransactionsRepositoryImpl.kt

```diff
@@ -0,0 +1,84 @@
+package org.p2p.wallet.history.interactor
+
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.withContext
+import org.p2p.solanaj.kits.transaction.SwapDetails
+import org.p2p.solanaj.kits.transaction.TransactionDetails
+import org.p2p.solanaj.model.types.AccountInfo
+import org.p2p.wallet.history.db.dao.TransactionDaoDelegate
+import org.p2p.wallet.history.model.HistoryTransaction
+import org.p2p.wallet.infrastructure.network.provider.TokenKeyProvider
+import org.p2p.wallet.rpc.repository.RpcRepository
+
+class HistoryTransactionsRepositoryImpl(
+    private val rpcRepository: RpcRepository,
+    private val tokenKeyProvider: TokenKeyProvider,
+    private val transactionDaoDelegate: TransactionDaoDelegate,
+    private val transactionDetailsMapper: TransactionDetailsMapper,
+    private val historyTransactionMapper: HistoryTransactionMapper
+) : HistoryTransactionsRepository {
+    override suspend fun getTransactionsHistory(
+        tokenPublicKey: String,
+        signatures: List<String>
+    ): List<HistoryTransaction> {
+        return withContext(Dispatchers.IO) {
```

Comment:
is it okay to do so?

Reply:
I am not sure if really need it. 
Do we need it?

Reply:
yep, we do need it.
Or all the code will run on MainThread